### PR TITLE
[HttpFoundation] Remove length limit on ETag

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/23766#discussion_r132072842
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by stof** [comment]
why is it done only in master ? In 3.4, you switched from a full hash to a truncated hash. So this is the place where your change should be merged.

**Comment by nicolas-grekas** [comment]
targeted

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/HttpFoundation/BinaryFileResponse.php [hunk 1]
```diff
@@ -141,7 +141,7 @@ public function setAutoLastModified()
      */
     public function setAutoEtag()
     {
-        $this->setEtag(substr(base64_encode(hash_file('sha256', $this->file->getPathname(), true)), 0, 32));
+        $this->setEtag(base64_encode(hash_file('sha256', $this->file->getPathname(), true)));
```
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/BinaryFileResponse.php]: You could remove the last padding char `=` that serves no purpose here.

**Pull request merged**